src/content/en/updates/2017/12/modulepreload.md

```diff
@@ -0,0 +1,147 @@
+First, the browser has to parse and compile it. Ideally,
+this should happen ahead of time as well, so that when the module is needed, the code is
+ready to run. However, V8 (Chrome's JavaScript engine) parses and compiles modules
+differently from other JavaScript. `<link rel="preload">` doesn't
+provide any way of indicating that the file being loaded is a module, so all the browser
+can do is load the file and put it in the cache. Once the script is loaded using a
+`<script type="module">` tag (or it's loaded by another module), the browser parses
+and compiles the code as a JavaScript module.
+
+## So is `<link rel="modulepreload">` just `<link rel="preload">` for modules?
+
+In a nutshell, yes. By having a specific `link` type for preloading modules, we can
+write simple HTML without worrying about what credentials mode we're using. The
+defaults just work.
+
+```html
+<head>
+  <link rel="modulepreload" href="super-critical-stuff.js">
+</head>
+[...]
+<script type="module" src="super-critical-stuff.js">
+```
+And since Chrome now knows that what you're preloading is a module, it can be smart
+and parse and compile the module as soon as it's done fetching, instead of waiting
+until it tries to run.
+
+## But what about modules' dependencies?
+
+Funny you should ask! There is indeed something we haven't talked about: recursivity.
+
+The `<link rel="modulepreload">` spec actually allows for optionally loading not just
+the requested module, but all of its dependency tree as well. Browsers don't have to
+do this, but they can.
+
+So what would be the best cross-browser solution for preloading a module and its
+dependency tree, since you'll need the full dependency tree to run the app?
+
+Browsers that choose to preload dependencies recursively should have robust deduplication
+of modules, so in general the best practice would be to declare the module and the flat list
+of its dependencies, and trust the browser not to fetch the same module twice.
+
+```html
+<head>
+  <!-- dog.js imports dog-head.js, which in turn imports
+       dog-head-mouth.js, and so on.  -->
+  <link rel="modulepreload" href="dog.js">
+  <link rel="modulepreload" href="dog-head.js">
+  <link rel="modulepreload" href="dog-head-mouth.js">
+  <link rel="modulepreload" href="dog-head-mouth-tongue.js">
+</head>
+```
+
+## Does preloading modules help performance?
+
+Preloading can help in maximizing bandwidth usage, by telling the browser about what
+it needs to fetch so that it's not stuck with nothing to do during those long roundtrips.
+If you're experimenting with modules and running into performance issues due to deep
+dependency trees, creating a flat list of preloads can definitely help!
+
+That said, there may be other performance issues with modules for now, so make sure
+you take a close look at what's happening in your application with Developer Tools, and
+consider bundling your application into several chunks in the meantime. There's plenty of
```

Comment:
I'm not sure we should recommend bundlers in this article without a more thorough explanation of their upsides and downsides, e.g. their impact on cache granularity.

Reply:
I agree that it would be nice to have further guidance, but it appears we're missing that on developers.google.com/web. That said, it's crucial for developers coding with modules to keep using bundlers for now, since they need to support non-module-enabled browsers, even if they're directly shipping code for module-enabled ones.